src/components/AddItemForm.js

```diff
@@ -43,7 +49,16 @@ class AddItemForm extends React.Component {
     });
   }
 
+  normalizeUserInput(item) {
+    item = item.toLowerCase().trim();
+    var pattern = /[^0-9a-z]/g;
+    item = item.replace(pattern, '');
+    return item;
+  }
+
   render() {
+    let normalizedInput = this.normalizeUserInput(this.state.name);
```

Comment:
Not a blocker; just something I realized while reading through this component.

Because the normalized input is being set in state as the user types, `this.state.name_normalized` is available and this method doesn't need to be called again.

`this.state.name_normalized` can be used instead on line 89: filter={['name_normalized', '==', this.state.name_normalized]}

Reply:
Wouldn't that be comparing the same thing though?? `this.state.name_normalized` is what is saved in state. The first parameter from the filter is also pulling the `name_normalized` from state. `normalizedInput` aka `this.normalizeUserInput(this.state.name)` is the user's input being run through the function as they are typing.

I did remove the variable and just use the function in the filer `{['name_normalized', '==', this.normalizeUserInput(this.state.name)]}`